src/components/People/People.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
+import { Person } from '../../types/Person';
+import { PersonItem } from '../PersonItem/PersonItem';
+import { SearchLink } from '../SearchLink';
+import { SortColumns } from '../../utils/SortColumns';
+import { getSortOrder } from '../../utils/functions';
+import { SortOrders } from '../../utils/SortOrders';
+
+const sortingLinks = [
+  { title: 'Name', value: SortColumns.NAME },
+  { title: 'Sex', value: SortColumns.SEX },
+  { title: 'Born', value: SortColumns.BORN },
+  { title: 'Died', value: SortColumns.DIED },
+];
+
+interface Props {
+  people: Person[],
+}
+
+export const People: React.FC<Props> = ({
+  people,
+}) => {
+  const [searchParams] = useSearchParams();
+
+  const sort = searchParams.get('sort') || '';
+  const order = searchParams.get('order') || '';
+
+  const findPerson = (name: string | null) => {
+    if (!name) {
+      return undefined;
+    }
+
+    return people.find(personItem => {
+      return personItem.name === name;
+    });
+  };
+
+  const handleOnSort = (sortColumn: SortColumns) => {
+    let newSortOrder: SortOrders | null = SortOrders.DEFAULT;
+
+    if (!sort || sort === sortColumn) {
+      newSortOrder = getSortOrder(order as SortOrders);
+    } else {
+      newSortOrder = SortOrders.DESC;
+    }
+
+    return { sort: newSortOrder ? sortColumn : null, order: newSortOrder };
+  };
```

Comment:
On the first click, you need to apply asc order (which means just sort by sortColumn), on the second click you need to apply descending order. But you have on the first click the correct order of people, but the wrong name of the order.